agent/consul/gateways/gateway_meta.go

```diff
@@ -0,0 +1,143 @@
+package gateways
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/consul/acl"
+	"github.com/hashicorp/consul/agent/consul/state"
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+// GatewayMeta embeds both a BoundAPIGateway and its corresponding APIGateway.
+type GatewayMeta struct {
+	// Bound is the bound config entry for a given gateway.
```

Comment:
Just for consistency
```suggestion
	// Bound is the bound-api-gateway config entry for a given gateway.
```